### PR TITLE
improve toString for core link resources

### PR DESCRIPTION
**Pull request by jvermillard**
Signed-off-by: Julien Vermillard <julien@vermillard.com>

### CODE REVIEW

File: leshan-core/src/main/java/org/eclipse/leshan/core/util/datatype/ValueToStringUtil.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        case OPAQUE:
+            // We don't print OPAQUE value as this could be credentials one.
+            // Not ideal but didn't find better way for now.
+            return ((byte[]) value).length + "Bytes";
+        case CORELINK:
+            return Arrays.toString((Link[]) value);
+        default:
+            return type.toString();
+        }
+    }
```
Comment (sbernard31) [inline leshan-core/src/main/java/org/eclipse/leshan/core/util/datatype/ValueToStringUtil.java:36]: Just an idea but Maybe better to call the class `LwM2mValueUtil` and the method `toPrettyString()` ? :thinking: 
(Or something like this)

This way we can image that `LwM2mValueUtil` could maybe be a place for future utility methods ?

Comment (jvermillard) [reply leshan-core/src/main/java/org/eclipse/leshan/core/util/datatype/ValueToStringUtil.java:36]: I had no idea how to name it, so I was pretty sure I'll bait you into finding one :joy:

File: leshan-core/src/main/java/org/eclipse/leshan/core/node/LwM2mResourceInstance.java [hunk 2]
```diff
@@ -224,7 +225,7 @@ public String toString() {
         // We don't print OPAQUE value as this could be credentials one.
         // Not ideal but didn't find better way for now.
         return String.format("LwM2mResourceInstance [id=%s, value=%s, type=%s]", id,
-                type == Type.OPAQUE ? ((byte[]) value).length + "Bytes" : value, type);
+                LwM2mValueUtil.toPrettyPrint(type, value), type);
```
Comment (sbernard31) [inline leshan-core/src/main/java/org/eclipse/leshan/core/node/LwM2mResourceInstance.java:228]: is `toPrettyPrint` instead of `toPrettyString` on purpose ?

Comment (jvermillard) [reply leshan-core/src/main/java/org/eclipse/leshan/core/node/LwM2mResourceInstance.java]: :facepalm:

**Review by sbernard31 — APPROVED** [review]
LGTM.

I will squash all commit in 1 and refer #1320 issue in commit header.

**Comment by sbernard31** [comment]
Integrated in master.

Thx @jvermillard :pray:

File: leshan-core/src/main/java/org/eclipse/leshan/core/util/datatype/LwM2mValueUtil.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    static public String toPrettyString(ResourceModel.Type type, Object value) {
+        switch (type) {
+        case OPAQUE:
+            // We don't print OPAQUE value as this could be credentials one.
+            // Not ideal but didn't find better way for now.
+            return ((byte[]) value).length + "Bytes";
+        case CORELINK:
+            return Arrays.toString((Link[]) value);
+        default:
+            return type.toString();
```
Comment (sbernard31) [inline leshan-core/src/main/java/org/eclipse/leshan/core/util/datatype/LwM2mValueUtil.java:34]: Oops : 
```diff
-          return type.toString();
+          return value.toString();
```
Fixed in 76dc3700479a68e5b6833f54750deede2b3f349d

**Pull request merged**